task5/func1.js

```diff
@@ -0,0 +1,21 @@
+class MyPromise {
```

Comment:
You could extend your class from Promise class and then you won't need a constructor and code could be reduced :)
`class MyPromise extends Promise`